pkg/flowkit/config/json/network.go

```diff
@@ -48,15 +49,17 @@ func transformNetworksToJSON(networks config.Networks) jsonNetworks {
 
 	for _, n := range networks {
 		jsonNetworks[n.Name] = jsonNetwork{
-			Host: n.Host,
+			Host:       n.Host,
+			NetworkKey: n.HostNetworkKey,
 		}
 	}
 
 	return jsonNetworks
 }
 
 type jsonNetwork struct {
-	Host string
+	Host       string
+	NetworkKey string
```

Comment:
I might be missing some complexity here, but for this case at least it seems like we can just add `json` tags to the struct, then wouldn't need the custom `Marshal`/`Unmarshal` methods. 
```suggestion
	Host       string `json:"host"`
	NetworkKey string `json:"network-key"`
```

Reply:
It's also a bug here since unmarshaling to `map[string]string` will always succeed which means advanced schema will be ignored. @jordanschalm is correct, tags could just be added (this code is pretty old), but at the same time like I mentioned you should add this key to the [advanced schema](https://github.com/onflow/flow-cli/blob/8ac4778e6653594cae8d94ba2c15ee01987b42e6/pkg/flowkit/config/json/network.go#L66), so it should just add another key and try [parse it here](https://github.com/onflow/flow-cli/blob/8ac4778e6653594cae8d94ba2c15ee01987b42e6/pkg/flowkit/config/json/network.go#L80).

Reply:
https://github.com/onflow/flow-cli/pull/449/commits/0e29e14541a6f2d948b2357a9b7ff9df72b5330b